src/nvim/drawscreen.c

```diff
@@ -1200,16 +1204,45 @@ void comp_col(void)
   set_vim_var_nr(VV_ECHOSPACE, sc_col - 1);
 }
 
-static void redraw_win_signcol(win_T *wp)
+static int buf_signcols_validate(win_T *wp, buf_T *buf, bool force)
```

Comment:
Should this go in the same file as `buf_signcols_validate_range`?

Reply:
Sure, both are only used in `drawscreen.c` but have more to do with decorations I guess. Would `static` `drawscreen.c` functions be preferred or exported in `decorations.c`?

Reply:
I think not `drawscreen.c`. I considered adding a new file just for column/signs, but if not that then maybe `decoration.c` for now.